examples/chef/common/chef-rvc-mode-delegate.cpp

```diff
@@ -197,14 +248,61 @@ void RvcCleanMode::Shutdown()
     }
 }
 
+chip::Protocols::InteractionModel::Status chefRvcCleanModeWriteCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                        const EmberAfAttributeMetadata * attributeMetadata,
+                                                                        uint8_t * buffer)
+{
+    // this cluster is only enabled for endpoint 1
+    if (endpointId != 1)
+    {
+        return chip::Protocols::InteractionModel::Status::UnsupportedEndpoint;
+    }
+
+    VerifyOrDie(gRvcCleanModeInstance != nullptr);
+
+    chip::Protocols::InteractionModel::Status ret;
+    ModeBase::Instance * clusterInstance = gRvcCleanModeInstance;
+    chip::AttributeId attributeId        = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::RvcCleanMode::Attributes::CurrentMode::Id: {
+        uint8_t m                                  = static_cast<uint8_t>(buffer[0]);
+        Protocols::InteractionModel::Status status = clusterInstance->UpdateCurrentMode(m);
+        if (Protocols::InteractionModel::Status::Success == status)
+        {
+            break;
+        }
+        ret = chip::Protocols::InteractionModel::Status::UnsupportedWrite;
```

Comment:
why do you replace the status from above with unsupportedwrite? why is this not `ret = status` ? A comment explaining why any failure status should go to unsupported write is needed here...